### PR TITLE
chore(orc8r): Switch Magma to proto V2

### PR DESCRIPTION
**Pull request by emakeev**
Signed-off-by: Evgeniy Makeev <evgeniym@fb.com>

## Summary

Switch Magma to proto V2

There is an effort of upgrading Magma to Go 1.18 (#12151), it includes 3 significant changes to codebase: 
1. proto upgrade
2. swagger upgrade
3. Go 1.18 upgrade

Each of the mentioned above changes are very large, so - authors, mainteners and reviewers could benefit from compartmentalizing them in separate PRs. This PR is the implementation of #1 - proto upgrade.

The proto upgrade involves significant changes to some of magma core APIs as well as unit tests. 
It makes sense to separate it, also - proto upgrade (or at least core of it) can be done independent of swagger & Go 1.18 upgrade.

The change provides proto V2 upgrade without over-reliance on relatively expensive proto clone calls.

The most significant required change is the modification of _**configurator APIs/interfaces**_ which were erroneously designed to pass & return config structures **by value**, the rest of the changes are mostly related to unit tests.

## Test Plan

fix & run unit tests

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/cloud/go/protos/oai/nas_state.pb.go [hunk 1]
```diff
@@ -3850,696 +4010,2150 @@ type EmmContext struct {
...
+// capability_s
+type EmmSecurityContext_Capability struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	EpsEncryption  uint32 `protobuf:"varint,1,opt,name=eps_encryption,json=epsEncryption,proto3" json:"eps_encryption,omitempty"`
+	EpsIntegirty   uint32 `protobuf:"varint,2,opt,name=eps_integirty,json=epsIntegirty,proto3" json:"eps_integirty,omitempty"`
+	UmtsEncryption uint32 `protobuf:"varint,3,opt,name=umts_encryption,json=umtsEncryption,proto3" json:"umts_encryption,omitempty"`
+	UmtsIntegirty  uint32 `protobuf:"varint,4,opt,name=umts_integirty,json=umtsIntegirty,proto3" json:"umts_integirty,omitempty"`
```
[the remark this answers is not in the document]
Comment (emakeev) [reply lte/cloud/go/protos/oai/nas_state.pb.go:4399]: the PR is strictly targets proto V2 upgrade, it's not intended to fixing unrelated existing spelling errors

File: lte/cloud/go/protos/oai/nas_state.pb.go [hunk 2]
```diff
@@ -3850,696 +4010,2150 @@ type EmmContext struct {
...
+}
+
+// capability_s
+type EmmSecurityContext_Capability struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	EpsEncryption  uint32 `protobuf:"varint,1,opt,name=eps_encryption,json=epsEncryption,proto3" json:"eps_encryption,omitempty"`
+	EpsIntegirty   uint32 `protobuf:"varint,2,opt,name=eps_integirty,json=epsIntegirty,proto3" json:"eps_integirty,omitempty"`
```
[the remark this answers is not in the document]
Comment (emakeev) [reply lte/cloud/go/protos/oai/nas_state.pb.go:4397]: the PR is strictly targets proto V2 upgrade, it's not intended to fixing unrelated existing spelling errors

[the remark this answers is not in the document]
Comment (emakeev) [reply lte/cloud/go/protos/oai/nas_state.pb.go:4397]: the PR is strictly targets proto V2 upgrade, it's not intended to fixing unrelated existing spelling errors

File: feg/cloud/go/protos/s6a_proxy.pb.go [hunk 3]
```diff
@@ -60,70 +79,91 @@ const (
...
+		"OUT_OF_SPACE":                    4002,
+		"ELECTION_LOST":                   4003,
+		"AUTHORIZATION_REJECTED":          5003,
+		"USER_UNKNOWN":                    5001,
+		"UNKNOWN_SESSION_ID":              5002,
+		"UNKNOWN_EPS_SUBSCRIPTION":        5420,
+		"RAT_NOT_ALLOWED":                 5421,
+		"ROAMING_NOT_ALLOWED":             5004,
+		"EQUIPMENT_UNKNOWN":               5422,
+		"UNKOWN_SERVING_NODE":             5423,
```
[the remark this answers is not in the document]
Comment (emakeev) [reply feg/cloud/go/protos/s6a_proxy.pb.go:137]: the PR is strictly targets proto V2 upgrade, it's not intended to fixing unrelated existing spelling errors

File: feg/cloud/go/protos/s6a_proxy.pb.go [hunk 4]
```diff
@@ -60,70 +79,91 @@ const (
...
+		4002: "OUT_OF_SPACE",
+		4003: "ELECTION_LOST",
+		5003: "AUTHORIZATION_REJECTED",
+		5001: "USER_UNKNOWN",
+		5002: "UNKNOWN_SESSION_ID",
+		5420: "UNKNOWN_EPS_SUBSCRIPTION",
+		5421: "RAT_NOT_ALLOWED",
+		5004: "ROAMING_NOT_ALLOWED",
+		5422: "EQUIPMENT_UNKNOWN",
+		5423: "UNKOWN_SERVING_NODE",
```
[the remark this answers is not in the document]
Comment (emakeev) [reply feg/cloud/go/protos/s6a_proxy.pb.go:109]: the PR is strictly targets proto V2 upgrade, it's not intended to fixing unrelated existing spelling errors

**Comment by maxhbr** [comment]
Hey @emakeev ,

I fully agree with your push towards simplified and split PRs. But please allow the original authors of #12151 to do and commit that, so that the attribution for that effort stays with the community members that invested the huge amount of work around the go upgrade.

**Comment by emakeev** [comment]
@maxhbr, NP - just trying to help, I suggested to split 12151 a while back (https://github.com/magma/magma/pull/12151#issuecomment-1093414274) & wasn't sure if there were issues with such approach.

File: feg/cloud/go/protos/s6a_proxy.pb.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (themarwhal) [reply feg/cloud/go/protos/s6a_proxy.pb.go:109]: @Neudrino I saw your other spelling fix PR :D Do you think we can fix these too? I *think* it's safe to rename protobuf fields as long as the type and number do not change

**Comment by Neudrino** [comment]
I think, this PR has now been superseded by https://github.com/magma/magma/pull/12546 and should be closed.

**Comment by emakeev** [comment]
closing since #12546 has been successfully merged

**Pull request closed without merging**